src/ReferenceList.php

```diff
@@ -211,12 +191,9 @@ public function removeReferenceHash( $referenceHash ) {
 	 * @return Reference|null
 	 */
 	public function getReference( $referenceHash ) {
-		/**
-		 * @var Hashable $hashable
-		 */
-		foreach ( $this as $hashable ) {
-			if ( $hashable->getHash() === $referenceHash ) {
-				return $hashable;
+		foreach ( $this->references as $reference ) {
+			if ( $reference->getHash() === $referenceHash ) {
```

Comment:
There are two loops in this class that search for a hash, but no method to access elements by index. Therefore it would probably be more efficient (less code, maybe faster) to index the array by hash, and not numerically.